SpaceHosting.Index/Sparnn/Distances/MatrixMetricSearchSpaceAlgorithm.cs

```diff
@@ -3,6 +3,7 @@ namespace SpaceHosting.Index.Sparnn.Distances
     internal enum MatrixMetricSearchSpaceAlgorithm
     {
         Cosine,
-        JaccardBinary
+        JaccardBinary,
+        JaccardBinarySingleFeatureOriented
```

Comment:
leading to performance measures, JaccardBinarySingleFeatureOriented usefull for k << n tasks, where k is the size of features to find, and n is cardinality of search space.
But if we move to cases k = const * n, the advantage is not so obvious. But i believe i know how to generalize suggested method to all cases; but its not reasonable to do it in this pull request

Reply:
It seems to me that huge search queries (k ~ n) is an extremely corner case. In practice one has millions of vectors in search space and is going to search for dozen or maybe hundred of nearest ones...

Besides if we are going to return that huge amounts of data we'll have a bunch of other concerns to tackle with. For instance we will have to change Search API and provide some kind of "cursor" to retrieve search results in batches since huge arrays of vectors will hit GC performance.

All that said I propose to leave a single `JaccardBinary` metric option in `MatrixMetricSearchSpaceAlgorithm` and just switch its implementation to new `JaccardBinarySingleFeatureOrientedSpace`. And we can  keep old `JaccardBinaryDistanceSpace` around internally for benchmarking and testing purposes.

Reply:
After running macro benchmarks for SparnnIndex we decided to just replace existing `JaccardBinaryDistanceSpace` implementation.